WS2012R2/lisa/setupscripts/FCOPY_utils.ps1

```diff
@@ -0,0 +1,167 @@
+		Write-Output "Info: hv_fcopy_daemon process is running."
+		$retValue = $True
+    }
+
+    del $filename
+    return $retValue
+}
+
+#######################################################
+#
+#	Stop hypervfcopyd or hv_fcopy_daemon when it is running on vm
+#
+#######################################################################
+function stop_fcopy_daemon()
+{
+    $sts = check_fcopy_daemon
+    if ($sts[-1] -eq $True ){
+        .\bin\plink -i ssh\${sshKey} root@${ipv4} "pkill -f 'fcopy'"
+        if (-not $?) {
+            Write-Error -Message  "ERROR: Unable to kill hypervfcopy daemon" -ErrorAction SilentlyContinue
+            Write-Output "ERROR: Unable to kill hypervfcopy daemon"
+            return $False
+        }
+    }
+    return $true
+}
+
+#######################################################################
+#
+#	Checks if test file is present
+#
+#######################################################################
+function check_file([String] $testfile)
+{
+    .\bin\plink -i ssh\${sshKey} root@${ipv4} "wc -c < /tmp/$testfile"
+    if (-not $?) {
+        Write-Output "ERROR: Unable to read file" -ErrorAction SilentlyContinue
+        return $False
+    }
+	return $True
+}
+
+#################################################################
+#
+# Remove file from vm
+#
+#################################################################
+function remove_file_vm() {
+    . .\setupScripts\TCUtils.ps1
+    $sts = SendCommandToVM $ipv4 $sshKey "rm -f /mnt/$testfile"
+    if (-not $sts) {
+        return $False
+    }
+    return $True
+}
+
+#################################################################
+#
+# Do fdisk /dev/sdb, mkfs /dev/sdb1 and mount /dev/sdb1 to /mnt
+#
+#################################################################
+function Mount-Disk()
+{
+    $driveName = "/dev/sdb"
+
+    $sts = SendCommandToVM $ipv4 $sshKey "(echo d;echo;echo w)|fdisk ${driveName}"
+    if (-not $sts) {
+        Write-Output "ERROR: Failed to format the disk in the VM $vmName."
+        return $Failed
+    }
+
+    $sts = SendCommandToVM $ipv4 $sshKey "(echo n;echo p;echo 1;echo;echo;echo w)|fdisk ${driveName}"
+    if (-not $sts) {
+        Write-Output "ERROR: Failed to format the disk in the VM $vmName."
+        return $Failed
+    }
+
+    $sts = SendCommandToVM $ipv4 $sshKey "mkfs.ext4 ${driveName}1"
+    if (-not $sts) {
+        Write-Output "ERROR: Failed to make file system in the VM $vmName."
+        return $Failed
+    }
+
+    $sts = SendCommandToVM $ipv4 $sshKey "mount ${driveName}1 /mnt"
+    if (-not $sts) {
+        Write-Output "ERROR: Failed to mount the disk in the VM $vmName."
+        return $Failed
+    }
+
+    "Info: $driveName has been mounted to /mnt in the VM $vmName."
+
+    return $True
+}
+
+#################################################################
+#
+# Check systemd available or not
+#
+#################################################################
+function Check-Systemd()
```

Comment:
let's get this function under TCUtils.ps1. it already exists into 2 other separate scripts, so we can consolidate it under tcutils.

Reply:
Done, thank you.